inspirehep/modules/workflows/utils/grobid_authors_parser.py

```diff
@@ -66,9 +66,9 @@ def __init__(self, author_selector):
     @staticmethod
     def _extract(source, path, type=None, text=False):
         if type:
-            path += "[@type='{type}']".format(type=type)
+            path += "[@type='{type}' and string-length(normalize-space()) > 0]".format(type=type)
         if text:
-            path += "/text()"
+            path += "[string-length(normalize-space()) > 0]/text()"
```

Comment:
won't this fail now if you provide both `type` and `text`? I'm not sure you can do `node[selector1][selector2]`.

Reply:
Actually It's working: 
```python 
> /home/pazembrz/dev/inspire-next/inspirehep/modules/workflows/utils/grobid_authors_parser.py(72)_extract()
     71             path += "[string-length(normalize-space()) > 0]/text()"
---> 72             return source.xpath(path)
     73         return source.xpath(path)

ipdb> path
"orgName[@type='institution' and string-length(normalize-space()) > 0][string-length(normalize-space()) > 0]/text()"
ipdb> source.xpath(path)
[<Selector xpath="orgName[@type='institution' and string-length(normalize-space()) > 0][string-length(normalize-space()) > 0]/text()" data=u'Bose National Centre for Basic Sciences'>, <Selector xpath="orgName[@type='institution' and string-length(normalize-space()) > 0][string-length(normalize-space()) > 0]/text()" data=u'JD Block'>]
```
https://www.w3.org/TR/1999/REC-xpath-19991116/#location-paths It's recognized as concatenation of those two selectors. But I have to agree that this is quite ugly as in both selectors one part is identical. I will change it a little.